db/db_impl.h

```diff
@@ -1678,6 +1678,12 @@ class DBImpl : public DB {
   size_t GetWalPreallocateBlockSize(uint64_t write_buffer_size) const;
   Env::WriteLifeTimeHint CalculateWALWriteHint() { return Env::WLTH_SHORT; }
 
+  static Status CreateWAL(DBImpl* impl, const EnvOptions& env_options,
```

Comment:
Curious, why do you need to pass in DBImpl? Asking because all these functions are anyway part of DBImpl class.

Reply:
never mind. I see that a new DBImpl is created inside ::Open, and moreover ::Open is static.